File: src/main/scala/io/gatling/custom/dummy/sampler/actions/DummyActionBuilder.scala
package io.gatling.custom.dummy.sampler.actions

import io.gatling.core.action.Action
import io.gatling.core.action.builder.ActionBuilder
import io.gatling.core.session.Expression
import io.gatling.core.structure.ScenarioContext

case class DummyActionBuilder(requestName: Expression[String], duration: Expression[Long]) extends ActionBuilder {
  override def build(ctx: ScenarioContext, next: Action): Action = new DummyRequestAction(requestName, duration, ctx, next)
}

File: src/main/java/io/gatling/custom/dummy/sampler/javaapi/actions/DummyActionBuilder.java
package io.gatling.custom.dummy.sampler.javaapi.actions;

import io.gatling.javaapi.core.ActionBuilder;
import io.gatling.javaapi.core.ChainBuilder;

public class DummyActionBuilder implements ActionBuilder {

    private final io.gatling.custom.dummy.sampler.actions.DummyActionBuilder wrapped;

    public DummyActionBuilder(io.gatling.custom.dummy.sampler.actions.DummyActionBuilder wrapped) {
        this.wrapped = wrapped;
    }

    @Override
    public io.gatling.core.action.builder.ActionBuilder asScala() {
        return wrapped;
    }

    @Override
    public ChainBuilder toChainBuilder() {
        return ActionBuilder.super.toChainBuilder();
    }
}
